jax/experimental/sparse/bcoo.py

```diff
@@ -572,6 +576,63 @@ def _bcoo_dot_general_abstract_eval(lhs_data, lhs_indices, rhs, *, dimension_num
   out_dtype = jnp.promote_types(lhs_data.dtype, rhs.dtype)
   return core.ShapedArray(out_shape, out_dtype)
 
+_bcoo_dot_general_default_translation_rule = xla.lower_fun(
+    _bcoo_dot_general_impl, multiple_results=False, new_style=True)
+
+def _bcoo_dot_general_cuda_translation_rule(
+    ctx, avals_in, avals_out, lhs_data, lhs_indices, rhs, *, dimension_numbers,
+    lhs_shape):
+
+  (lhs_contract, rhs_contract), _ = dimension_numbers
+  _, lhs_indices_aval, _, = avals_in
+  _, n_sparse, _, _ = _validate_bcoo_indices(lhs_indices_aval, lhs_shape)
```

Comment:
minor: it's slightly cleaner I think to do `props = _validate_bcoo_indices(...)` and then refer to `props.n_sparse` below.